src/Tools/AnalyzerRunner/Program.cs

```diff
@@ -0,0 +1,480 @@
+            return Task.FromResult(sum);
+        }
+
+        private static async Task<Statistic> GetAnalyzerStatisticsAsync(Document document, CancellationToken cancellationToken)
+        {
+            SyntaxTree tree = await document.GetSyntaxTreeAsync(cancellationToken).ConfigureAwait(false);
+
+            SyntaxNode root = await tree.GetRootAsync(cancellationToken).ConfigureAwait(false);
+
+            var tokensAndNodes = root.DescendantNodesAndTokensAndSelf(descendIntoTrivia: true);
+
+            int numberOfNodes = tokensAndNodes.Count(x => x.IsNode);
+            int numberOfTokens = tokensAndNodes.Count(x => x.IsToken);
+            int numberOfTrivia = root.DescendantTrivia(descendIntoTrivia: true).Count();
+
+            return new Statistic(numberOfNodes, numberOfTokens, numberOfTrivia);
+        }
+
+        private static IEnumerable<DiagnosticAnalyzer> FilterAnalyzers(IEnumerable<DiagnosticAnalyzer> analyzers, Options options)
+        {
+            foreach (var analyzer in analyzers)
+            {
+                if (options.UseAll)
+                {
+                    yield return analyzer;
+                }
+                else if (options.AnalyzerIds.Count == 0)
+                {
+                    if (analyzer.SupportedDiagnostics.Any(i => i.IsEnabledByDefault))
+                    {
+                        yield return analyzer;
+                    }
+
+                    continue;
+                }
+                else if (analyzer.SupportedDiagnostics.Any(y => options.AnalyzerIds.Contains(y.Id)))
+                {
+                    yield return analyzer;
+                }
+            }
+        }
+
+        private static ImmutableArray<Assembly> GetAnalyzerAssemblies(string path)
+        {
+            if (File.Exists(path))
+            {
+                return ImmutableArray.Create(Assembly.LoadFrom(path));
+            }
+            else if (Directory.Exists(path))
+            {
+                var builder = ImmutableArray.CreateBuilder<Assembly>();
+                foreach (var file in Directory.GetFiles(path, "*.dll", SearchOption.AllDirectories))
+                {
+                    builder.Add(Assembly.LoadFrom(file));
+                }
+
+                return builder.ToImmutable();
+            }
+
+            Utilities.WriteLine($"Cannot load assembliy or assemblies from {path}.", ConsoleColor.Red);
+            return ImmutableArray<Assembly>.Empty;
+        }
+
+        private static IEnumerable<Type> GetLoadableTypes(Assembly assembly)
+        {
+            if (assembly == null) throw new ArgumentNullException(nameof(assembly));
+            try
+            {
+                return assembly.GetTypes();
+            }
+            catch (ReflectionTypeLoadException e)
+            {
+                Console.WriteLine($"Failed to load all types from {assembly.FullName}. Will proceed with types loaded successfully.");
```

Comment:
it could be nice if it prints out loader exceptions so that users can know why types failed to be loaded.

Reply:
Removed this code and use AnalyzerLoader.

---
In reply to: [149862165](https://github.com/dotnet/roslyn/pull/23087#discussion_r149862165) [](ancestors = 149862165)